### PR TITLE
Add amd metrics support

### PR DESCRIPTION
**Pull request by Binyang2014**
Add AMD GPU metrics support.
Currently, we will export `GPU usage`, `GPU memory usage`, `GPU temperature` for AMD GPU

### CODE REVIEW

File: src/job-exporter/src/docker_inspect.py [hunk 1]
```diff
@@ -75,8 +75,11 @@ def parse_docker_inspect(inspect_output):
             # for kube-launcher tasks
             if k == "FC_TASK_INDEX":
                 m["PAI_TASK_INDEX"] = v
-            elif k == "NVIDIA_VISIBLE_DEVICES" and v != "all" and v != "void":
-                m["GPU_ID"] = v
+            else:
+                if k == "NVIDIA_VISIBLE_DEVICES" and v and v != "all" and v != "void":
+                    m["GPU_ID"] = v
+                if k == "PAI_AMD_VISIBLE_DEVICES" and v:
```
Comment (abuccts) [inline src/job-exporter/src/docker_inspect.py]: those two env vars exist at the same time, change the second `if` to `elif`?

File: src/job-exporter/test/data/docker_inspect_amd.json [hunk 2]
```diff
@@ -0,0 +1,337 @@
...
+                "Type": "bind",
+                "Source": "/var/lib/kubelet/pods/a1310f10-5ea6-11ea-8a2a-90b11c27f535/etc-hosts",
+                "Destination": "/etc/hosts",
+                "Mode": "",
+                "RW": true,
+                "Propagation": "rprivate"
+            }
+        ],
+        "Config": {
+            "Hostname": "srgsora-05",
```
Comment (abuccts) [inline src/job-exporter/test/data/docker_inspect_amd.json]: change a name

File: src/job-exporter/test/data/docker_inspect_amd.json [hunk 3]
```diff
@@ -0,0 +1,337 @@
...
+            "CpuRealtimeRuntime": 0,
+            "CpusetCpus": "",
+            "CpusetMems": "",
+            "Devices": [
+                {
+                    "PathOnHost": "/dev/fuse",
+                    "PathInContainer": "/dev/fuse",
+                    "CgroupPermissions": "rwm"
+                }
+            ],
```
Comment (abuccts) [inline src/job-exporter/test/data/docker_inspect_amd.json]: if this a real amd docker?  no gpu devices here

File: docs/alerting/README.md [hunk 4]
```diff
@@ -48,8 +48,8 @@ Other pai component also used some metrics for display, they are:
     <td>
         <ul>
             <li>node_uname_info</li>
-            <li>nvidiasmi_utilization_gpu</li>
-            <li>nvidiasmi_utilization_memory</li>
+            <li>gpu_utilization</li>
+            <li>gpu_mem_utilization</li>
```
Comment (abuccts) [inline docs/alerting/README.md]: I prefer the name **`gpu_util`** and **`memory_usage`**, according to pretty printed `nvidia-smi` output.
util is a percentage, while usage is a value of XX MB, there's also a `memory_util` field in smi xml output.

File: src/job-exporter/src/amd.py [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+    for index, value in enumerate(values):
+        gpu_util = float(value["GPU use (%)"])
+        gpu_mem_vram_total = float(value["vram Total Memory (B)"])
+        gpu_mem_vram_used = float(value["vram Total Used Memory (B)"])
+        gpu_mem_vram_util = gpu_mem_vram_used / gpu_mem_vram_total * 100
+        gpu_temperature = float(value["Temperature (Sensor edge) (C)"])
+        gpu_uuid = str(value["Unique ID"]).strip()
+        pci_addr = value["PCI Bus"]
+        res[str(index)] = AMDGpuStatus(gpu_util, gpu_mem_vram_util, gpu_uuid,
+                                       gpu_temperature, pci_addr)
```
Comment (abuccts) [inline src/job-exporter/src/amd.py]: try catch in parsing and assign default values? what if a field is missing in a certain situation

File: src/job-exporter/src/docker_inspect.py (hunk 1 above)
Comment (abuccts) [reply src/job-exporter/src/docker_inspect.py]: or decide by `get_gpu_vendor()`?

Comment (abuccts) [reply src/job-exporter/src/docker_inspect.py]: and what if `NVIDIA_VISIBLE_DEVICES` is `all` or `void`, will skip this env?
pls also add the [`none` case](https://github.com/NVIDIA/nvidia-container-runtime#nvidia_visible_devices).

File: src/job-exporter/src/amd.py (hunk 5 above)
Comment (Binyang2014) [reply src/job-exporter/src/amd.py]: Here is a `try-catch` sentence to catch the exception during collecting. If a field is missing, an exception will raise. And we will abort the metrics in this iteration.
https://github.com/microsoft/pai/blob/90604120dfee205a5312ae0b591deb0a0c0bc48e/src/job-exporter/src/collector.py#L235-L245

File: src/job-exporter/test/data/docker_inspect_amd.json (hunk 3 above)
Comment (Binyang2014) [reply src/job-exporter/test/data/docker_inspect_amd.json]: An exception will raise during run `rocm-smi`. The related metrics will not export to `Prometheus`

File: docs/alerting/README.md (hunk 4 above)
Comment (Binyang2014) [reply docs/alerting/README.md]: Here `gpu_mem_utilization` is a percentage. In Grafana, we only show the percentage of GPU memory utilization, not the usage values

File: src/job-exporter/src/docker_inspect.py (hunk 1 above)
Comment (Binyang2014) [reply src/job-exporter/src/docker_inspect.py]: Sure, those two env vars will exist at the same time. But only one will has the value. If all contain the value, it should be mis-config. Will add log for this situation.

In previous logic, if `NVIDIA_VISIBLE_DEVICES` is `all` or `void`, will skip this env. I don't change the logic here. Will add `none` here.

Comment (Binyang2014) [reply src/job-exporter/src/docker_inspect.py]: For AMD GPU, seems both `NVIDIA_VISIBLE_DEVICES` and `PAI_AMD_VISIBLE_DEVICES` will be set....
Will correct this.

**Comment by Binyang2014** [comment]
Test case:
1. Deploy new job-exporter/grafana in the bed which contains AMD node.
2. Submit AMD job. Check if node gpu metric, job gpu metric can show correctly

**Pull request merged**